### DISCUSSION TITLE
How to check if Pointer is PRESSED

### DISCUSSION THREAD

**Discussion by trashhacker** [AvaloniaUI/Avalonia#17164] [Q&A] [answered]
how can i check the current pointer Pressed state in PointerMoved Handler?

or can someone tell me why the PointerPressed event is not fired for DataGridRow´s
this is a bug right ?

**Comment by LaurentInSeattle** [comment]
You should probably use PointerPointProperties available in the Event Args for PointerMoved. 
See: 
https://reference.avaloniaui.net/api/Avalonia.Input/PointerEventArgs/
and: 
https://reference.avaloniaui.net/api/Avalonia.Input/PointerPointProperties/ 

HTH

**Comment by stevemonaco** [comment, marked as answer]
Check the Events tab in DevTools. It's possible that `DataGrid` or `DataGridRow` handles the event before you do which means you need to use `AddHandler` to create a handler for tunneling events or for already-handled events. You can only do this from code.

**Comment by trashhacker** [comment]
like that ? but i dont get hook

            if (AssociatedObject is InputElement input)
            {
                input.AddHandler(InputElement.PointerReleasedEvent, PointerReleasedHandler);
                input.AddHandler(InputElement.PointerPressedEvent, PointerPressedHandler);
                input.AddHandler(InputElement.PointerMovedEvent, PointerMovedHandler);
            }

ah got it, thanks yea DataGridRow eats the event

                input.AddHandler(InputElement.PointerPressedEvent, PointerPressedHandler, RoutingStrategies.Direct | RoutingStrategies.Tunnel | RoutingStrategies.Bubble, true);
